beacon-chain/rpc/prysm/v1alpha1/validator/proposer_attestations.go

```diff
@@ -309,3 +317,211 @@ func (vs *Server) deleteAttsInPool(ctx context.Context, atts []ethpb.Att) error
 	}
 	return nil
 }
+
+// isAttestationFromCurrentEpoch returns true if the attestation is from the current epoch.
+func (vs *Server) isAttestationFromCurrentEpoch(att ethpb.Att) bool {
+	currentSlot := vs.TimeFetcher.CurrentSlot()
```

Comment:
Is this prone to race conditions? The current  slot here is computed with the wall clock and then we take slots and epochs from the data and forkchoice. What happens if the slot changes while evaluating these functions? (eg what happens if there's a clock drift backwards of a few milliseconds since the block proposal was called and this function is hit).

Reply:
Given that this is only used for the current epoch, we are safe until slot 0. Typically, in slot 0, we don't call get block until 100ms because the get duty call

Reply:
maybe we can just fetch the current slot once, and provide it as an argument to both functions